ui/src/app/utils/index.ts

```diff
@@ -217,3 +217,17 @@ export class ReactWrapperBase implements DoCheck, OnInit, OnDestroy {
     );
   }
 }
+
+export function decamelize(str: string, separator: string) {
+  separator = typeof separator === 'undefined' ? '_' : separator;
+
+  return str
+      .replace(/([a-z\d])([A-Z])/g, '$1' + separator + '$2')
+      .replace(/([A-Z]+)([A-Z][a-z\d]+)/g, '$1' + separator + '$2')
+      .toLowerCase();
+}
+
+export function capitalize(s: string) {
+  return s.toLowerCase().replace( /\b./g,
+      function(a) { return a.toUpperCase(); } );
+}
```

Comment:
`lodash` has several string functions that might suffice here. Specifically see `lowerCase` and `capitalize`